project/adcf3016/README.md

```diff
@@ -0,0 +1,103 @@
+        - Choosing the median from each dimension.
+        And then compare which dimension's median makes the number of left and right subtrees closer to each other, as better as possible.
+        This will create balanced k-d tree.
+        - We have the following parameters that allow us to determine the different options for choosing the root
+            - cell
+            - bounding rectangle
+            - aspect ratio
+            - spread
+
+    2. Implement the basic operation of k-d tree such as `findNearest` and `rangeSearch`
+
+## Prospective Users
+- A simple library of k-d tree for those who need to store and access data with multiple key values, and also need to search for a specific range of data.
+
+## API Description
+- k-d tree class overview
+    ```c++
+    class KdTree {
+        public:
+            kdTree(vector<vector<double>> data_set, int dimension);
+            void insert(vector<double> node);
+            bool delete(vector<double> node);
+            bool search(vector<double> node);
+            vector<double> findNearest(vector<double> node);
+            vector<vector<double>> rangeSearch(vector<double> lowerbound,
+                                               vector<double> upperbound);
+            vector<double> findMin(int dimension);
+            vector<double> findMax(int dimension);
+
+        private:
+            KdTree *left;
+            KdTree *right;
+            vector<double> key;
+    };
+    ```
+- class function usage
+    - `kdTree`
+        - Choosing the root node and recursive build subtree
+    - `insert`
+        - Insert new data into current k-d tree
+    - `delete`
+        - Delete the provided data in k-d tree.
+        Return delete sucess or not.
+    - `search`
+        - Return if the data provided is in the tree or not
+    - `findNearest`
+        - Find the nearest node to the node provided
+    - `rangeSearch`
+        - Find all data with key value between lowerbound and upperbound
+    - `findMin` & `findMax`
+        - Find min/max value of the given dimension
+
+## Engineering Infrastructure
+1. Automatic build system: `Makefile`
+2. Version control: `git`
+3. Wrapping module: `pybind11`
+4. Testing framework: `pytest`
+5. Documentation: `README.md`
+
+## Schedule
+### Week 1 (10/30)
+- Study different types of k-d tree splits and decide what type to implement.
+- Implement `kdTree`
+- Write the Makefile for this project
+
+### Week 2 (11/6)
+- Continue implement `kdTree`
+
+### Week 3 (11/13)
+- Implement `insert`, `delete`, `search`
+
+### Week 4 (11/20)
+- Implement `findMin`, `findMax`
+- Write pytest to test `kdTree`, `insert`, `delete`, `findMin`, `findMax`
+
+### Week 5 (11/27)
+- Implement `findNearest`, `rangeSearch`
+
+### Week 6 (12/4)
+- Write pytest to test `findNearest`, `rangeSearch`
+
+### Week 7 (12/11)
+- Flextime
+
+### Week 8 (12/18)
+- Flextime
+
+## References

```

Comment:
There should be at least one paper describing the algorithm.  It's a good idea to list it here.